node/src/bead/mod.rs

```diff
@@ -65,15 +57,41 @@ impl Decodable for Bead {
     }
 }
 
+#[repr(u8)]
+#[derive(Debug, Clone, Copy)]
+pub enum BeadRequestType {
+    GetBeads = 0,
+    GetTips = 1,
+    GetGenesis = 2,
+    GetAllBeads = 3,
+    GetBeadsAfter = 4,
+    BeadResponseError = 5,
+}
```

Comment:
Add `impl Encodable for BeadSyncError` and `impl Decodable for BeadSyncError`

Please do that for all enums that are used as network messages.